<launch>
    <arg name="bag_file" default="" />
    <arg name="camera_file" default="$(find omni_slam_eval)/launch/default_camera.yaml" />
    <arg name="results_file" default="$(eval ''.join(arg('bag_file').split('.')[:-1]) + '.' + ''.join(arg('camera_file').split('/')[-1].split('.')[:-1]) + '.tracking.hdf5')" />
    <arg name="rate" default="1" />
    <node pkg="omni_slam_eval" type="omni_slam_tracking_eval_node" name="omni_slam_tracking_eval_node" required="true" output="screen">
        <param name="bag_file" value="$(arg bag_file)" />
        <param name="results_file" value="$(arg results_file)" />
        <param name="image_topic" value="/unity_ros/Camera/FisheyeCamera/image_raw" />
        <param name="depth_image_topic" value="/unity_ros/Camera/FisheyeDepthCamera/image_raw" />
        <param name="pose_topic" value="/unity_ros/Camera/TrueState/pose" />
        <param name="tracked_image_topic" value="/omni_slam/tracked" />
        <param name="rate" value="$(arg rate)" />
        <rosparam subst_value="true">
            detector_type: 'GFTT'
            detector_parameters: {maxCorners: 100, qualityLevel: 0.01, minDistance: 5, blockSize: 5}
            tracker_type: 'lk'
            tracker_window_size: 128
            tracker_num_scales: 4
            tracker_checker_epipolar_threshold: 0.008
            tracker_checker_iterations: 1000
            tracker_delta_pixel_error_threshold: 3.0
            tracker_error_threshold: 20.0
            min_features_per_region: 10
            max_features_per_region: 999999
            keyframe_interval: 1
        </rosparam>
    </node>
    <rosparam command="load" file="$(arg camera_file)" ns="omni_slam_tracking_eval_node" />
</launch>
